bigquery_etl/view/__init__.py

```diff
@@ -190,16 +190,28 @@ def _valid_view_naming(self):
             for t in parsed.tokens
             if not (t.is_whitespace or isinstance(t, sqlparse.sql.Comment))
         ]
-        is_view_statement = (
-            " ".join(tokens[0].normalized.split()) == "CREATE OR REPLACE"
-            and tokens[1].normalized == "VIEW"
+        is_view_statement = " ".join(tokens[0].normalized.split()) in (
+            "CREATE",
+            "CREATE OR REPLACE",
+        ) and (
+            tokens[1].normalized == "VIEW"
+            or " ".join(t.normalized for t in tokens[1:3]).upper()
+            == "MATERIALIZED VIEW"
         )
         if is_view_statement:
-            target_view = str(tokens[2]).strip().split()[0]
-            try:
-                [project_id, dataset_id, view_id] = target_view.replace("`", "").split(
-                    "."
+            view_kw_index = 1 if tokens[1].normalized == "VIEW" else 2
+            if (
+                " ".join(
+                    t.normalized for t in tokens[view_kw_index + 1 : view_kw_index + 4]
                 )
+                == "IF NOT EXISTS"
+            ):
+                view_id_token = tokens[view_kw_index + 4]
+            else:
+                view_id_token = tokens[view_kw_index + 1]
+            target_view = str(view_id_token).replace("`", "").strip().split()[0]
+            try:
+                [project_id, dataset_id, view_id] = target_view.split(".")
```

Comment:
This took me a minute to parse. Not sure if some comments might make sense here?

Reply:
Yeah, I was annoyed how verbose this logic ended up being.  I was hoping we could just grab the first identifier token, but unfortunately sqlparse currently misclassifies the `MATERIALIZED` keyword as an identifier.

Actually, since this isn't a pressing issue, I think I'll wait for them to fix the bug I submitted, then this can be simplified to just look for the first identifier token.

Or an alternate approach would be using sqlparse just to remove comments from the SQL, then using regular expressions to extract the view ID.

Reply:
> Or an alternate approach would be using sqlparse just to remove comments from the SQL, then using regular expressions to extract the view ID.

I've implemented this alternate approach.